engine/src/flutter/lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -1535,8 +1535,25 @@ abstract class DefaultTextEditingStrategy
     event as DomFocusEvent;
 
     final DomElement? willGainFocusElement = event.relatedTarget as DomElement?;
-    if (willGainFocusElement == null ||
-        _viewForElement(willGainFocusElement) == activeDomElementView) {
+    if (willGainFocusElement == null) {
+      // If the element to gain focus is reported as `null`, it means that the
+      // window/iframe that Flutter runs in is losing focus. In this case, the
+      // engine signals to the framework to close the text input connection,
+      // allowing the focus to move elsewhere.
+      //
+      // This is fixing https://github.com/flutter/flutter/issues/155265.
+      textEditing.sendTextConnectionClosedToFrameworkIfAny();
+    } else if (_viewForElement(willGainFocusElement) == activeDomElementView) {
+      // If the focus stays within the same FlutterView, ensure the focus stays
+      // on the input element.
+
+      // TODO(166857): the motivation/reasoning behind this remains murky.
```

Comment:
Please, follow [the style guide](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-checking-in-comments-that-ask-questions) for flutter TODOs:

> TODOs should include the string TODO in all caps, followed by the GitHub username of the person with the best context about the problem referenced by the TODO in parentheses. A TODO is not a commitment that the person referenced will fix the problem, it is intended to be the person with enough context to explain the problem. Thus, when you create a TODO, it is almost always your username that is given.

```suggestion
      // TODO(yjbanov): Make text input less grabby. See: https://github.com/flutter/flutter/issues/166857
      // The motivation/reasoning behind whay this is needed is murky.
```